File: 12월 3주차/호텔 방배정.py
import sys
sys.setrecursionlimit(200000)
ht = {}
def find(x):
    if not ht.get(x):
        ht[x] = x+1
    else:
        ht[x] = find(ht[x])
        
    return ht[x]
    
def solution(k, room_number):
    answer = []
    for i in room_number:
        answer.append(find(i)-1)
    return answer

File: 5월 1주차/호텔 방배정.js
function find(x,rooms){
    if (rooms.has(x)){
        rooms.set(x,find(rooms.get(x),rooms))
        return rooms.get(x)
    }
    rooms.set(x,x+1)
    return x
}


function solution(k, room_number) {
    var answer = [];
    const rooms = new Map()
    room_number.forEach(room=>{
        let checkIn = find(room,rooms)
        answer.push(checkIn)
    })
    
    
    return answer;
}